### ISSUE TITLE
Dev site returning odd result for the example DOI on the front page

### ISSUE THREAD

**Issue by markmacgillivray** [OAButton/discussion#1636]
dev site returning odd result for the example DOI on the front page - investigate and fix asap. Likely something to do with new parallelisation

**Comment by markmacgillivray** [comment]
@JosephMcArthur this is just an update to let you know things have been done. I have investigated this and am still checking a few things. So far, it appears that some older dev records that were searched using the url or the citation end up creating catalogue records that don't match to the searches by ID. So when you later search for the same thing you will get back an "open" response if the previous ID search is pulled from the catalogue, or you get the "requested" response, which then leads to a closed request due to the item age. This is all technically correct, apart from that the searches without IDs should be getting matched into the same catalogue records as the ones with IDs, so I am looking at how to improve that. Also, I will add some checks to see if there is more than one catalogue record, and if so, try to pick the best one rather than just the most recent one.

I've been doing this along with doing some permissions API work. I'll write an update on that soon.

**Comment by markmacgillivray** [comment]
This has been fixed/improved. Altering the algorithm we use to try to keep things as fast as possible, but also match as many records as possible to ones we already found. Particularly, when a URL search was done then a citation search was done, for the same thing, it was hard to get them to identify as the same article, because those two inputs cause quite different orders of activity on the new parallel backend. However, it has been improved now without having noticeable impact in the speed improvements we had already gained, and the demo ones on the front page of the site do now all return the same record from our catalogue and subsequent searches for the same thing update that record rather than creating separate ones with possible differences in the result they display. It took a bit longer than hoped, but well worth it for deploying the new approach anyway.
